### PR TITLE
ENH: support variable copyright dates

### PR DESCRIPTION
**Pull request by ebolyen**

### CODE REVIEW

**Comment by ebolyen** [comment]
Tested on q2lint a couple of ways, example of output:
```
Invalid header: q2lint/_main.py (incorrect copyright years or does not match reference)
```

```
Invalid LICENSE file (incorrect copyright years or does not match reference)
Invalid header: q2lint/_main.py (incorrect copyright years or does not match reference)
```

plain mismatch:
```
Invalid LICENSE file (does not match reference)
Invalid header: q2lint/_main.py (does not match reference)
```

File: q2lint/_main.py [hunk 1]
```diff
@@ -7,8 +7,28 @@
...
 import pathlib
 import sys
+import re
+
+
+YEAR_PLACEHOLDER = "COPYRIGHT_YEARS"  # chosen to avoid re.escape replacement
+
+
+def check_license(license, reference, copyright_idx):
+    for idx, (line, ref) in enumerate(zip(license.splitlines(),
```
Comment (jairideout) [inline q2lint/_main.py]: I think we should use `itertools.zip_longest` so that the loop doesn't terminate early if one license is shorter than the other.

**Review by jairideout — CHANGES_REQUESTED** [review]
Looks great @ebolyen! Just had one comment.

File: q2lint/_main.py (hunk 1 above)
Comment (ebolyen) [reply q2lint/_main.py]: Good catch!

**Review by jairideout — APPROVED** [review]
Thanks @ebolyen!

**Pull request merged**